strings/intern_test.go

```diff
@@ -0,0 +1,34 @@
+package strings
+
+import (
+	. "gopkg.in/check.v1"
+	"testing"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type InternStringSuite struct {
+	pool *InternStringPool
+}
+
+func (i *InternStringSuite) SetUpTest(c *C) {
+	i.pool = NewInternStringPool()
+}
+
+var _ = Suite(&InternStringSuite{})
+
+func (i *InternStringSuite) TestIntern(c *C) {
+	// Takes a string as parameter and stores it to the string pool
+	i.pool.Intern("foo string")
+	r, ok := i.pool.get("foo string")
+	c.Assert(ok, Equals, true)
+	c.Assert(r, Equals, "foo string")
```

Comment:
Could you verify two copies of the same intern string have to the same Data pointer and Len (i.e., reflect.StringHeader are identical)?  Conversely, two copies of the same non-intern string should not have the same Data pointer.

Reply:
Hello. I tried this, but even for non-intern strings the Data of stringheader end up being identical.

Reply:
the golang compiler also performs string intern-ing for simple constants.  e.g., in https://play.golang.org/p/qSEMMzPZrMk y and z shares the same Data, but not x

Reply:
there was a typo.  this is the corrected example https://play.golang.org/p/R5njXXk30zk

Reply:
Thank you! This was helpful. I've tried various different methods of string generation and I am getting strings with distinct with stringheader data now :). strings.join, bytes array and concatenation are all generating different Data representations for the string.